adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -647,13 +647,13 @@ func createRoute(params *routeCreateParams) *routev3.Route {
 			Value: true,
 		},
 	}
-	if apiType == model.WS {
-		decorator = &routev3.Decorator{
-			Operation: endpointBasepath,
-		}
-	} else if apiType == model.HTTP || apiType == model.WEBHOOK {
+	// route path could be empty only if there is no basePath for API or the endpoint available,
+	// and resourcePath is also an empty string.
+	// Empty check is added to run the gateway in failsafe mode, as if the decorator string is
+	// empty, the route configuration does not apply.
+	if strings.TrimSpace(routePath) != "" {
```

Comment:
if `ws://localhost:8080` is the endpoint. can routePath be empty?

Reply:
nope. Since the APIs hosted in gateway will have a context all the time.